static/src/js/actions/__tests__/subscriptions.test.js

```diff
@@ -156,9 +155,11 @@ describe('createSubscription', () => {
         }
       })
 
-    useEdscStore.setState((state) => {
-      // eslint-disable-next-line no-param-reassign
-      state.collection.collectionId = 'collectionId'
+    useEdscStore.setState({
+      collection: {
+        ...useEdscStore.getState().collection,
+        collectionId: 'collectionId'
+      }
```

Comment:
Even with the eslint disable I prefer the previous way of setting the state. When doing `state.collection.collectionId = ` we are letting the immer library ensure that the rest of the state remains the same, which you are having to do manually with `...useEdscStore.getState().collection,`.

For a simple field like the collectionId that isn't bad, but it gets hard to maintain when you have a deeper nested object. 

Once these actions are moved to Zustand the eslint disable won't be necessary, because we ignore that rule for those directories for this reason.